controllers/operandbindinfo/operandbindinfo_controller.go

```diff
@@ -559,6 +668,40 @@ func sanitizeOdlmRouteData(m map[string]string, route ocproute.RouteSpec) (map[s
 	return sanitized, nil
 }
 
+// sanitizeServiceData takes a map, i.e. ODLM's Service.Data, and a K8s Service object
+// and returns a map ready to be included into a ConfigMap's data. The ODLM's
+// Service.Data is sanitized because the values are YAML fields in a K8s Service.
+// Ensures that:
+//  1. the field actually exists, otherwise returns an error
+//  2. extracts the value from the K8s Service's field, the value will be
+//     stringified
+func sanitizeServiceData(m map[string]string, service corev1.Service) (map[string]string, error) {
+	sanitized := make(map[string]string, len(m))
+	trueValue := ""
+	jpath := jsonpath.New("sanitizeServiceData")
+	for k, v := range m {
+		stringParts := strings.Split(v, "+")
+		for _, s := range stringParts {
+			actual := s
+			if strings.HasPrefix(s, ".") {
+				if len(s) > 1 {
+					if err := jpath.Parse("{" + s + "}"); err != nil {
+						return nil, err
+					}
+					buf := new(bytes.Buffer)
+					if err := jpath.Execute(buf, service); err != nil {
+						return nil, err
+					}
+					actual = buf.String()
+				}
+			}
+			trueValue += actual
+		}
+		sanitized[k] = trueValue
```

Comment:
@bitscuit 

Do we need to reset `trueValue` for every iteration?

Reply:
yes, good catch